kolibri/core/api.py

```diff
@@ -263,12 +263,12 @@ def serialize_object(self, **filter_kwargs):
         try:
             filter_kwargs = filter_kwargs or self._get_lookup_filter()
             return self.serialize(queryset.filter(**filter_kwargs))[0]
-        except IndexError:
+        
+        except (IndexError, ValueError, TypeError):
             raise Http404(
                 "No %s matches the given query." % queryset.model._meta.object_name
             )
 
-
```

Comment:
And it also doesn't like this removed line (it requires two blank lines betwee class/function definitions).